gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/DichotomyParallelization.java

```diff
@@ -35,31 +40,35 @@ public DichotomyParallelization(DichotomyLogging dichotomyLogging, DichotomyRunn
     }
 
     public SweResponse launchDichotomy(SweData sweData) {
-        final ExecutionResult executionResult = ParallelExecution
+        final ExecutionResult<SweDichotomyResult> executionResult = ParallelExecution
                 .of(() -> runDichotomyForOneDirection(sweData, DichotomyDirection.ES_FR))
                 // .and(() -> runDichotomyForOneDirection(sweData, Direction.FR_ES))
                 // .and(() -> runDichotomyForOneDirection(sweData, Direction.ES_PT))
                 // .and(() -> runDichotomyForOneDirection(sweData, Direction.PT_ES))
                 .close();
         dichotomyLogging.logEndAllDichotomies();
-        String ttcDocUrl = outputService.buildAndExportTtcDocument(sweData, executionResult);
-        SweResponse sweResponse = new SweResponse(sweData.getId(), ttcDocUrl);
         // build swe response from every response
-        return sweResponse;
+        String ttcDocUrl = outputService.buildAndExportTtcDocument(sweData, executionResult);
+        String voltageEsFrZipUrl = outputService.buildAndExportEsFrVoltageDoc(sweData, executionResult);
+        return  new SweResponse(sweData.getId(), ttcDocUrl, voltageEsFrZipUrl);
     }
 
     SweDichotomyResult runDichotomyForOneDirection(SweData sweData, DichotomyDirection direction) {
         DichotomyResult<RaoResponse> dichotomyResult = dichotomyRunner.run(sweData, direction);
         dichotomyLogging.logEndOneDichotomy(direction);
         // Generate files specific for one direction (cne, cgm, voltage) and add them to the returned object (to create)
+        Optional<VoltageMonitoringResult> voltageMonitoringResult = runVoltageCheck(sweData, dichotomyResult, direction);
         // fill response for one dichotomy
-        SweDichotomyResult sweDichotomyResult = new SweDichotomyResult(buildReturnedMap(direction, dichotomyResult));
-        return sweDichotomyResult;
+        return new SweDichotomyResult(direction, dichotomyResult, voltageMonitoringResult);
     }
 
-    private EnumMap<DichotomyDirection, DichotomyResult<RaoResponse>> buildReturnedMap(DichotomyDirection direction, DichotomyResult<RaoResponse> dichotomyResult) {
-        EnumMap<DichotomyDirection, DichotomyResult<RaoResponse>> mapResult = new EnumMap<>(DichotomyDirection.class);
-        mapResult.put(direction, dichotomyResult);
-        return mapResult;
+    private Optional<VoltageMonitoringResult> runVoltageCheck(SweData sweData, DichotomyResult<RaoResponse> dichotomyResult, DichotomyDirection direction) {
```

Comment:
Can you move out this method ? I dont think that it belongs to this class

Reply:
sure in a seperate sevice maybe ?